File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.py
class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        dupes = []
        for i in range(len(nums)):
            if nums[abs(nums[i])-1] < 0:
                dupes.append(abs(nums[i]))
            else:
                nums[abs(nums[i]) - 1] *= -1
        return dupes
        

File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findDuplicates = function(nums) {
  dupes = []
  
  dupes = nums.filter((number,i) => {
    if(nums[Math.abs(number)-1] > 0){
      nums[Math.abs(number)-1] *= -1
      return false
    }else{
      return true
    }
  }).map(el => Math.abs(el));
    
  return dupes
};